cli/azd/cmd/pipeline.go

```diff
@@ -55,493 +46,64 @@ For more information, go to https://aka.ms/azure-dev/pipeline.`,
-	if err != nil {
-		return false, err
-	}
-
-	if ghActionsInUpstreamRepo {
-		// upstream is already listing GitHub actions.
-		// There's no need to check if there are local workflows
-		return false, nil
+	// Read or init env
+	console := input.GetConsole(ctx)
+	if console == nil {
+		log.Panic("missing input console in the provided context")
 	}
 
-	// Upstream has no GitHub actions listed.
-	// See if there's at least one workflow file within .github/workflows
-	ghLocalWorkflowFiles := false
-	defaultGitHubWorkflowPathLocation := filepath.Join(
-		azdCtx.ProjectDirectory(),
-		".github",
-		"workflows")
-	err = filepath.WalkDir(defaultGitHubWorkflowPathLocation,
-		func(folderName string, file fs.DirEntry, e error) error {
-			if e != nil {
-				return e
-			}
-			fileName := file.Name()
-			fileExtension := filepath.Ext(fileName)
-			if fileExtension == ".yml" || fileExtension == ".yaml" {
-				// ** workflow file found.
-				// Now check if this file is already tracked by git.
-				// If the file is not tracked, it means this is a new file (never pushed to mainstream)
-				// A git untracked file should not be considered as GitHub workflow until it is pushed.
-				newFile, err := gitCli.IsUntrackedFile(ctx, azdCtx.ProjectDirectory(), folderName)
-				if err != nil {
-					return fmt.Errorf("checking workflow file %w", err)
-				}
-				if !newFile {
-					ghLocalWorkflowFiles = true
-				}
-			}
-
-			return nil
-		})
-
+	env, err := loadOrInitEnvironment(ctx, &p.manager.RootOptions.EnvironmentName, azdCtx, console)
 	if err != nil {
-		return false, fmt.Errorf("Getting GitHub local workflow files %w", err)
+		return fmt.Errorf("loading environment: %w", err)
 	}
 
-	if ghLocalWorkflowFiles {
-		message := fmt.Sprintf("\n%s\n"+
-			" - If you forked and cloned a template, please enable actions here: %s.\n"+
-			" - Otherwise, check the GitHub Actions permissions here: %s.\n",
-			output.WithHighLightFormat("GitHub actions are currently disabled for your repository."),
-			output.WithHighLightFormat("https://github.com/%s/actions", repoSlug),
-			output.WithHighLightFormat("https://github.com/%s/settings/actions", repoSlug))
-
-		console.Message(ctx, message)
-
-		rawSelection, err := console.Select(ctx, input.ConsoleOptions{
-			Message: "What would you like to do now?",
-			Options: []string{
-				manualChoice.String(),
-				cancelChoice.String(),
-			},
-			DefaultValue: manualChoice,
-		})
-
-		if err != nil {
-			return false, fmt.Errorf("prompting to enable github actions: %w", err)
-		}
-		choice := gitHubActionsEnablingChoice(rawSelection)
+	// TODO: Providers can be init at this point either from azure.yaml or from command args
+	// Using GitHub by default for now. To be updated to either GitHub or Azdo.
+	// The CI provider might need to have a reference to the SCM provider if its implementation
+	// will depend on where is the SCM. For example, azdo support any SCM source.
+	p.manager.ScmProvider = &pipeline.GitHubScmProvider{}
```

Comment:
Longer term, I am wondering if it would be clearer to not have manager be a state variable on the action and instead a local, we create and initialize here with all the state.  Or perhaps at that point the manager is returned by some function that does the `azure.yaml` processing.

Reply:
The only benefit of having it as state variable is that we can write the input arguments from the command directly into the manager options.  Otherwise, I would need to introduce local state that is later used to create the manager

Reply:
Makes sense